From f2a460cdf7ab7ab26ad7c121e2bbf9eb21dfcbd0 Mon Sep 17 00:00:00 2001
From: Dom Rodriguez <shymega@shymega.org.uk>
Date: Wed, 20 May 2026 13:23:26 +0100
Subject: [PATCH] fix: Scrap problematic `lock_cmd` `hypridle` script

Signed-off-by: Dom Rodriguez <shymega@shymega.org.uk>
---
 homes/configs/programs/hyprland.nix | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/homes/configs/programs/hyprland.nix b/homes/configs/programs/hyprland.nix
index c767224..bbab0b4 100644
--- a/homes/configs/programs/hyprland.nix
+++ b/homes/configs/programs/hyprland.nix
@@ -281,18 +281,7 @@ in {
         in
           lib.getExe cmd;
       in {
-        lock_cmd = let
-          cmd = pkgs.writeShellScriptBin "hyprlock_wrapper" ''
-            pidof hyprlock >/dev/null
-            if [ "$?" -eq 1 ]; then
-              ${pkgs.hyprlock}/bin/hyprlock
-            else
-              pkill -9 hyprlock
-              $0
-            fi
-          '';
-        in
-          lib.getExe cmd;
+        lock_cmd = lib.getExe pkgs.hyprlock;
         on_lock_cmd = "hyprctl dispatch dpms off && ${media-pause}";
         on_unlock_cmd = "hyprctl dispatch dpms on && ${pkgs.wireplumber}/bin/wpctl set-mute @DEFAULT_AUDIO_SINK@ 0";
         before_sleep_cmd = "loginctl lock-session";
-- 
2.51.2

